circleguard/visualizer.py

```diff
@@ -119,6 +123,32 @@ def __init__(self, replays=[], beatmap_id=None, beatmap_path=None, parent=None,
         self.setAutoFillBackground(True)
         self.setPalette(pal)
 
+    def resizeEvent(self, event):
+        width = (GAMEPLAY_WIDTH + GAMEPLAY_PADDING_WIDTH * 2)
+        height = (GAMEPLAY_HEIGHT + GAMEPLAY_PADDING_HEIGHT * 2)
+        y_scale = event.size().height() / height
+        x_scale = event.size().width() / width
+        if width * y_scale > event.size().width():
+            self.scale = x_scale
+            self.y_offset = (self.size().height() - height * x_scale) / 2
+            self.x_offset = 0
+        else:
+            self.scale = y_scale
+            self.y_offset = 0
+            self.x_offset = (self.size().width() - width * y_scale) / 2
+
+    def _x(self, position):
+        return self.x_offset + self.scaled_number(GAMEPLAY_PADDING_WIDTH + position)
+
+    def _y(self, position):
+        return self.y_offset + self.scaled_number(GAMEPLAY_PADDING_HEIGHT + position)
+
+    def scaled_point(self, x, y):
+        return QPointF(self._x(x), self._y(y))
+
+    def scaled_number(self, n):
```

Comment:
rename to `scale_point` and `scale_number`

Reply:
actually this could kinda go either way, it's a transformation on an object so passive tense works

Reply:
I like the passive version more since it applies an already defined scale. (you aren't passing a new scale)

Reply:
good point